src/nethsec/firewall/__init__.py

```diff
@@ -690,6 +690,27 @@ def get_zone_by_name(uci, name: str) -> str:
             return (z, zones[z])
     return (None, None)
 
+def get_rule_by_name(uci, name: str, tag = "") -> str:
+    """
+    Get rule config name by rule name, optionally filtered by tag.
+    Assume there is only one rule with the same name.
+
+    Args:
+        uci: EUci pointer
+        name: rule name
+        tag: optional tag to filter rules
+
+    Returns:
+        tuple of rule config name and rule config if rule with name name exists, (None, None) otherwise
+    """
+    rules = utils.get_all_by_type(uci, 'firewall', 'rule')
+    for r in rules:
+        if uci.get('firewall', r, 'name', default='') == name:
+            if not tag or (tag and tag in uci.get('firewall', r, 'ns_tag', default=[])):
```

Comment:
```suggestion
            if not tag or (tag in uci.get('firewall', r, 'ns_tag', default=[])):
```

Reply:
Current condition is a little bit different and more efficient: tag is not searched on the list if it is None.
But ok to simplify it a bit